proposals/nnnn-tuple-of-value-pack-expansion.md

```diff
@@ -0,0 +1,88 @@
+
+### Distinction between tuple values and value packs
+
+The following example demonstrates a pack repetition pattern on a value pack and an abstract tuple value separately first, then together but with the repetition pattern operating only on the value pack, and finally with the repetition pattern operating on both the value pack and the tuple value's contained value pack together interleaved.
+
+```swift
+func example<each U>(packElements value: repeat each U, tuple: (repeat each U)) {
+  print((repeat each value))
+  print((repeat each tuple))
+  
+  print((repeat (each value, tuple)))
+  
+  print((repeat (each value, each tuple)))
+}
+
+example(packElements: 1, 2, 3, tuple: (4, 5, 6))
+
+// Prints the following output:
+// (1, 2, 3)
+// (4, 5, 6)
+// ((1, (4, 5, 6)), (2, (4, 5, 6)), (3, (4, 5, 6)))
+// ((1, 4), (2, 5), (3, 6))
+```
+
+## Detailed design
+
+Pack reference expressions inside a repetition pattern can have abstract tuple type. The outer structure of the tuple is removed, leaving the elements of a value pack:
+
+```swift
+func expand<each T>(value: (repeat each T)) -> (repeat (each T)?) {
+  return (repeat each value)
+}
+```
+
+Applying the pack repetition pattern effectively removes the outer structure of the tuple leaving just the value pack. In the repetition expression, the base tuple is evaluated once before iterating over its elements.
+
+```swift
+repeat each <tuple expression>
+
+// the above is evaluated like this
+let tempTuple = <tuple expression>
+repeat each tempTuple
+```
+TODO: is this example actually informative or illustrative?
+
+## Source compatibility
+
+There is no source compatibility impact given that this is an additive change. It enables compiling code that previously would not compile.
+
+## ABI compatibility
+
+This proposal does not add or affect ABI as its impact is only on expressions. It does not change external declarations or types. It rests atop the ABI introduced in the **Value and Type Parameter Packs** proposal.
+
+## Implications on adoption
+
+Given that this change rests atop the ABI introduced in the **Value and Type Parameter Packs** proposal, this shares with it the same runtime back-deployment story.
+
+## Alternatives considered
+
+The **Value and Type Parameter Packs** proposal provides no means for declaring local variable value packs or stored property packs. This is mentioned as a potential future direction, but without that, containing the packs within tuples is the way to access such functionality. Therefore an alternative to this proposal would be to instead pursue those future directions instead. However that would only partially address the motivations for this proposal, still leaving a broader pack repetition pattern design needed for abstract tuples.
```

Comment:
This sounds like it's not really an alternative to the proposal but rather a potentially orthogonal future direction?

An alternative here might be just the originally implemented syntax (which some users in the pitch stage said they favored)--that way, you've got an opportunity to explain/defend why your chosen syntax proposed here is better.